src/main/g8/src/main/scala/$package$/DemoParallelism.scala

```diff
@@ -1,48 +1,75 @@
 package $package$
 
+import cats.effect.Resource
 import com.typesafe.scalalogging.LazyLogging
 import monix.eval.Task
 import monix.execution.Scheduler
+import monix.reactive.Observable
 
-import scala.concurrent.Await
 import scala.concurrent.duration._
 import scala.util.Random
 
 object DemoParallelism extends LazyLogging {
 
   /**
-    * Demonstrates effective parallelism using Monix and Monte-Carlo pi calculation.
-    *
-    * @param iterations  Iterations to calculate pi for
-    * @param parallelism Concurrent processes
-    */
-  def apply(iterations: Long, parallelism: Int): Unit = {
-    logger.info(f"Will approximate Pi with \$iterations%,d iterations of random bubble-filling and \$parallelism%,d parallel processes...")
+   * Demonstrates effective parallelism using Monix and Monte-Carlo pi calculation.
+   *
+   * @param iterations  Iterations to calculate pi for
+   * @param parallelism Concurrent processes
+   */
+  def apply(iterations: Long, parallelism: Int): Task[Double] = for {
+    // log that the work is starting
+    _ <- Task {
+      logger.info(f"Will approximate Pi with \$iterations%,d iterations of random bubble-filling and up to \$parallelism%,d parallel processes...")
+    }
 
-    // Scheduler for parallelism:
-    implicit val s: Scheduler = Scheduler.forkJoin(parallelism, maxThreads = 1000)
+    // create scheduler resource based on desired parallelism (it will be closed once we have finished our calculations)
+    schedulerResource = Resource.make(acquire = Task {
+      Scheduler.forkJoin(parallelism = parallelism, maxThreads = 1000)
+    })(release = s => Task { s.shutdown() })
 
-    // Approximation of pi using iteration:
-    val pi = approximatePi(iterations).runToFuture
+    // then use the scheduler to...
+    calculate <- schedulerResource.use(scheduler => {
+      // calculate an approximation of pi
+      approximatePi(parallelism, iterations)
+        // measure how long it took to perform the calculation
+        .timed
+        // set a timeout of 30 seconds
+        .timeout(30.seconds)
+        // insure the process runs on the provided scheduler
```

Comment:
```suggestion
        // ensure the process runs on the provided scheduler
```
Yeah, I'm that kind of guy.